config/testdata/secret.go

```diff
@@ -0,0 +1,42 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+)
+
+type Input struct {
+	Version string   `json:"version"`
+	Secrets []string `json:"secrets"`
+}
+
+type SecretOutput struct {
+	Value string `json:"value"`
+	// Use pointer for error so if it's not provided null will be used
```

Comment:
I believe you can add `omitempty` (or something similar) to the json tag to achieve this without using a pointer

Reply:
I think the behavior is different `omitempty` will return a json like `{"value": "abc"}` when we want `{"value": "abc", "error": null}`